Frontend/student-hub/src/App.js

```diff
@@ -13,10 +13,15 @@ import Dashboard from './pages/Dashboard';
 import CoursePage from './pages/CoursePage';
 import Profile from './pages/Profile';
 import MentorshipPage from './pages/MentorshipPage';
+import FriendsList from './pages/Friends';
+import { Snackbar, Alert } from '@mui/material';
+
 
 function App() {
 
   const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [open, setOpen] = useState(false);
```

Comment:
can we make this name more descriptive? what is open?

Reply:
This is a variable name for the M-UI that I used for the snackbar. It was the prop variable name for the bool.

Reply:
got it! now that I'm looking at it again, do you ever call `setError` with something nonnull without calling `setOpen(true)`?

instead of having two variables to control the visibility state, maybe you can just pass in `open={error !== null}`?

Reply:
Can you explain this a little more detail for me. Thanks